lib/pages/search.dart

```diff
@@ -10,11 +20,152 @@ class SearchPage extends StatefulWidget {
 }
 
 class _SearchPageState extends State<SearchPage> {
+  String searchQuery = '';
+  SortType sortType = SortType.name;
+  bool sortAscending = true;
+  bool showPastEvents = false;
+
+  late Future<QuerySnapshot<Map<String, dynamic>>> firestoreQuery;
+  late TextEditingController controller;
+
+  Future<QuerySnapshot<Map<String, dynamic>>> getFirestoreCollection() {
+    return FirebaseFirestore.instance
+        .collection('events')
+        .where(
+          "date",
+          isGreaterThan: showPastEvents ? null : DateTime.timestamp().millisecondsSinceEpoch,
+        )
+        .get();
+  }
+
+  List<Event> filterResults(List<Event> events) {
+    return events
+        .where(
+          (element) =>
+              element.name.toLowerCase().contains(searchQuery) ||
+              element.description.toLowerCase().contains(searchQuery),
+        )
+        .toList()
+      ..sort((a, b) {
+        int result = 0;
+        switch (sortType) {
+          case SortType.name:
+            result = a.name.compareTo(b.name);
+            break;
+          case SortType.date:
+            result = a.date.compareTo(b.date);
+            break;
+          // case SortType.location:
+          //   result = 0; //dodam kiedyś
+        }
+        return sortAscending ? result : -result;
+      });
+  }
+
+  @override
+  void initState() {
+    controller = TextEditingController();
+    firestoreQuery = getFirestoreCollection();
+    super.initState();
+  }
+
   @override
   Widget build(BuildContext context) {
-    return const Scaffold(
-      appBar: AppHeader(),
-      body: Text('Search Page'),
+    return Scaffold(
+      appBar: const AppHeader(),
+      body: Padding(
+        padding: const EdgeInsets.all(8.0),
+        child: Column(
+          children: [
+            Row(
+              children: [
+                Expanded(
+                  child: TextField(
+                    controller: controller,
+                    decoration: const InputDecoration(hintText: "Search"),
+                    autocorrect: false,
+                    onSubmitted: (value) {
+                      setState(() {
+                        searchQuery = value.toLowerCase();
+                      });
+                    },
+                  ),
+                ),
+                Switch(
+                  value: showPastEvents,
+                  onChanged: (value) {
+                    setState(() {
+                      showPastEvents = value;
+                      firestoreQuery = getFirestoreCollection();
+                    });
+                  },
+                ),
+                const Text("Past Events"),
+              ],
+            ),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
```

Comment:
ja bym je polaczyl

Reply:
Jak? Wszystkie kombinacje listowac? Imo głupie jeśli nie dodam sortowania po dystansie jak chciałem to może